src/api/app/models/project.rb

```diff
@@ -532,6 +528,20 @@ def fill_repo_cache
   def update_one_repository_add_pathes(repo)
     current_repo = self.repositories.find_by_name(repo['name'])
 
+    # sync download on demand config
+    hasharray=[]
+    repo.elements('download') { |d|
+      hash = { arch: d['arch'], url: d['url'], repotype: d['repotype'],
+               archfilter: d['archfilter'], pubkey: d['pubkey']
+             }
+      if d['master']
+        hash[:masterurl] = d['master']['url']
+        hash[:mastersslfingerprint] = d['master']['sslfingerprint']
+      end
+      hasharray << hash
+    }
+    sync_hash_with_model(DownloadRepository, current_repo.download_repositories, hasharray)
```

Comment:
Something like

```ruby
download_repositories = []
repo.elements('download').each do |download|
  download_repository = DownloadRepository.new(arch: download['arch'],
                                               url: download['url'],
                                               repotype: download['repotype'],
                                               archfilter: download['archfilter'],
                                               pubkey: download['pubkey'])
  if download['master']
     download_repository.masterurl = download['master']['url']
     download_repository.mastersslfingerprint = download['master']['sslfingerprint']
  end
  download_repositories << download_repository
end

current_repo.download_repositories.replace(download_repositories)
```
.replace will perform a diff and delete/add only records that have changed.